elastic4s-client-esjava/src/main/scala/com/sksamuel/elastic4s/http/JavaClient.scala

```diff
@@ -128,6 +128,9 @@ object JavaClient extends Logging {
 
     val client = RestClient
       .builder(hosts: _*)
+      .setPathPrefix(
+        props.endpoints.map(_.prefix).collectFirst { case Some(prefix) => prefix }.getOrElse("")
```

Comment:
would only work if all of the prefixes are the same (or all absent), would it make sense to fail with an exception otherwise?

Reply:
I think that's an unusual edge case, but we could fail yes.